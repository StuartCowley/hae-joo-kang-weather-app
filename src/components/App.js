import React, { useEffect, useState } from "react";
import "../styles/App.css";
import axios from "axios";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";

const App = () => {
  const [location, setLocation] = useState({ city: "", country: "" });
  const [forecasts, setForecasts] = useState([]);
  const [selectedDate, setSelectedDate] = useState(0);

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => setSelectedDate(date);

  const getForecast = () => {
    const endpoint = "https://mcr-codes-weather-app.herokuapp.com/forecast";

    axios.get(endpoint).then((response) => {
      setSelectedDate(response.data.forecasts[0].date);
      setLocation(response.data.location);
      setForecasts(response.data.forecasts);
    });
  };

  useEffect(() => {
    getForecast();
  });

  return (
    <div className="weather-app">
      <LocationDetails city={location.city} country={location.country} />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
};

export default App;
