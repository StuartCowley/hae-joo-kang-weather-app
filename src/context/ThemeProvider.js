/* eslint-disable react/jsx-no-constructed-context-values */
import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const ThemeContext = createContext();

export const useTheme = () => useContext(ThemeContext);

export const ThemeProvider = ({ children }) => {
  const [darkTheme, setDarkTheme] = useState(false);
  return (
    <ThemeContext.Provider value={{ darkTheme, setDarkTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};
ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
