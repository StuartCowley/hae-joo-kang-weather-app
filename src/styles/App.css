.weather-app {
  margin: 0 20px;
  background-image: url('../../images/daytime_sky.jpg');
  background-repeat: no repeat;
  background-size: cover;
  background-position: center top;
  text-align: center;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(8px + 1.5vmin);
  color: black;
}
