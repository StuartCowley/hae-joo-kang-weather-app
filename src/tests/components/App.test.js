import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../../components/App";
import { ThemeProvider } from "../../context/ThemeProvider";

describe("App", () => {
  it("renders App correctly", () => {
    const { asFragment } = render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
