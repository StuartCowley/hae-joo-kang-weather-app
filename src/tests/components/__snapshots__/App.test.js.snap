// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<div
  className="weather-app light"
>
  <div
    className="search-form"
  >
    <input
      data-testid="search-form__input"
      onChange={[Function]}
      placeholder="city name"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </div>
  <h2
    className="location"
  >
    , 
  </h2>
  <div
    className="forecast-summaries"
  />
  <div
    className="toggle-container"
  >
    <input
      id="switch"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="switch"
    >
      ToggleTheme
    </label>
  </div>
</div>
`;
